raft/raft.go

```diff
@@ -828,6 +828,9 @@ func (r *raft) campaign(t CampaignType) {
 		}
 		sort.Slice(ids, func(i, j int) bool { return ids[i] < ids[j] })
 	}
+
+	commit := r.raftLog.committed
+	commitTerm := r.raftLog.zeroTermOnErrCompacted(r.raftLog.term(commit))
```

Comment:
I think it should panic.

Reply:
It may possible that `r.raftLog.committed == r.raftLog.applied` and all log were compacted

Reply:
I don't think so. Valid range for querying term is `[first index - 1, last index]`. In the case you described, `applied index = compact index = first index - 1`.